portal/views/assessment_engine.py

```diff
@@ -847,14 +848,20 @@ def assessment_update(patient_id):
             'valid': True,
         })
 
-    # Todo: enforce identifier uniqueness at initial submission
+    try:
+        identifier = Identifier.from_fhir(updated_qnr.get('identifier'))
+    except (TypeError, KeyError):
+        abort(400, "Ill formed or missing 'identifier', see FHIR spec")
     try:
         existing_qnr = QuestionnaireResponse.query.filter(
-            QuestionnaireResponse.document["identifier"]
-            == updated_qnr["identifier"]
-        ).one()
-    # except NoResultException:
+            QuestionnaireResponse.document['identifier']['system']
+            == json.dumps(identifier.system)).filter(
```

Comment:
Could you move this `filter(` to its own line so it's obvious it's chained off the previous `filter()`?

Reply:
those are all over the code base, easy to parse once you get used to reading them :)